enumerable_methods.rb

```diff
@@ -0,0 +1,63 @@
+# Project 2: Enumerable Methods
+
+module Enumerable
+  def my_each(&proc)
+    length.times do |i|
+      proc.call(self[i])
+    end
+    self
+  end
+
+  def my_each_with_index(&proc)
+    length.times do |item, _i|
+      proc.call(item)
+    end
+    self
+  end
+
+  def my_select(&proc)
+    selection = []
+    length.times do |i|
+      selection.push(self[i]) if proc.call(self[i])
+    end
+    selection
+  end
+
+  def my_all?(&proc)
+    length.times do |i|
+      return false if proc.call(self[i]) == false
+    end
+  end
+
+  def my_any?(&proc)
+    length.times do |i|
+      return true if proc.call(self[i]) == true
+    end
+  end
```

Comment:
* [ ] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_any? returns true if the block ever returns a value other than false or nil
  ```ruby
    array = [1,2,5,4,7]
    true_block = proc {|num| num < 10 }
    p array.my_any?(&true_block) == array.any?(&true_block) # false
  ```

* [ ] my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
  ```ruby
    array = [nil, false, true, []]
    p array.my_any? == array.any? # false
  ```
* [ ] my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class
  ```ruby
    array = [1,2,5,4,7]
    p array.my_any?(Integer) == array.any?(Integer) # false
  ```
* ( ] my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
  ```ruby
    array = %w[dog door rod blade]
    p array.my_any?(/c/) == array.any?(/c/) # false
  ```
* [ ] my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
  ```ruby
    array = %w[dog door rod blade]
    p array.my_any?(5) == array.any?(5) # false
  ```